subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -1633,8 +1633,15 @@ static void light_ctrl_srv_reset(struct bt_mesh_model *model)
 {
 	struct bt_mesh_light_ctrl_srv *srv = model->user_data;
 	struct bt_mesh_light_ctrl_srv_cfg cfg = BT_MESH_LIGHT_CTRL_SRV_CFG_INIT;
+#if CONFIG_BT_MESH_LIGHT_CTRL_SRV_REG
+	struct bt_mesh_light_ctrl_srv_reg_cfg reg_cfg = BT_MESH_LIGHT_CTRL_SRV_REG_CFG_INIT;
+#endif
 
 	srv->cfg = cfg;
+#if CONFIG_BT_MESH_LIGHT_CTRL_SRV_REG
+	srv->reg.cfg = reg_cfg;
```

Comment:
This seems fine. On the other hand, seems like `srv->reg.cfg` remains uninitialized when the device is provisioned for the very first time and on subsequent bootups if the light CTRL has not happened to store the regulator data yet. In such cases `lc_setup_srv_settings_set()` will never be called.

Reply:
Regulator is always initialized on bootup by `BT_MESH_LIGHT_CTRL_SRV_INIT`.